tests/benchmark/unit/src/test_c_allocators.cpp

```diff
@@ -0,0 +1,174 @@
+#include <iostream>
+
+#include <gtest/gtest.h>
+
+#include "memory_management.hpp"
+#include "test_config.hpp"
+#include "test_types.hpp"
+
+TEST(CAllocators, MallocReallocFree)
+{
+    CUSTOM_MEMORY_MANAGER.ResetStatistics();
```

Comment:
it would be group unit test code into 3 blocks, with the following comments on top 

// Arrange

// Act

// Assert

Reply:
I saw that in the mesh kernel unit tests. Nicely organized. 

In my opinion, in the current context, segregating assertions/expectations and putting them in the end will devalue the test. Assertions and expectations are performed every time an object's memory is modified until the object is freed. 

What do you think?